modules/lvgl/input/lvgl_pointer_input.c

```diff
@@ -90,14 +78,53 @@ static void lvgl_pointer_process_event(const struct device *dev, struct input_ev
 		point->x = cap->x_resolution - point->y;
 		point->y = tmp;
 	}
+}
+
+static void lvgl_pointer_process_event(const struct device *dev, struct input_event *evt)
+{
+	const struct lvgl_pointer_input_config *cfg = dev->config;
+	struct lvgl_common_input_data *data = dev->data;
+	lv_disp_t *disp = lv_disp_get_default();
+	struct lvgl_disp_data *disp_data = disp->driver->user_data;
+	struct display_capabilities *cap = &disp_data->cap;
+	lv_point_t *point = &data->pending_event.point;
+
+	switch (evt->code) {
+	case INPUT_ABS_X:
+		if (cfg->swap_xy) {
+			point->y = evt->value;
+			invert_y_if_needed(dev, point);
+		} else {
+			point->x = evt->value;
+			invert_x_if_needed(dev, point);
+		}
+		rotate_point_if_needed(dev, point);
+		break;
+	case INPUT_ABS_Y:
+		if (cfg->swap_xy) {
+			point->x = evt->value;
+			invert_x_if_needed(dev, point);
+		} else {
+			point->y = evt->value;
+			invert_y_if_needed(dev, point);
+		}
+		rotate_point_if_needed(dev, point);
```

Comment:
wouldn't this end up doing `rotate_point_if_needed` twice? hard to wrap my head around the logic. How about having a handle_x() and handle_y(), passing the new point as an argument and then doing all the check for that point in the function with no swapping involved? Maybe you could handle both invert and rotation in one shot without going to the `point` field for all the intermediate step but just set it at the end.

Reply:
yep, the `rotate_point_if_needed` would be called twice, so not correct. I tried implementing your suggestion but the issue that I was facing is that the rotation can only happen after the `sync` since it involves the x and y dimension of the point which are only "set" after the event sync. Filtering step needs to happen after the rotation, since it moves the coordinates around. The only thing you could move into the switch is the inversion if I got it right.

Reply:
Can't you duplicate the rotation code and handle x and y individually?

Reply:
Say we have the current point `(x,y)` and the display dimensions given as `(w,h)` and the display is rotated by 90°. Rotating the point involves both of its dimensions:
```
if (cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_90) {
	lv_coord_t tmp;
	tmp = point->x;
    point->x = point->y;
    point->y = cap->y_resolution - tmp;
}
```
We can only perform the rotation once the x *and* y dimension are set, otherwise the result is dependant on the order of the incoming events:
```
State:
x = 0 (not assigned yet)
y = 0 (not assigned yet)

Incoming INPUT_ABS_X event:
x = 5
Performing the rotation as above:
x = 0, y = h - 5

Incoming INPUT_ABS_Y event:
y = 2 
Performing the rotation as above:
x = 2, y = h
```
So all in all the coordinate of the resulting point appear to be `(2, h)` which is incorrect. 
If you get the events in the opposite order:
```
State:
x = 0 (not assigned yet)
y = 0 (not assigned yet)

Incoming INPUT_ABS_Y event:
y = 2 
Performing the rotation as above:
x = 2, y = h

Incoming INPUT_ABS_X event:
x = 5
Performing the rotation as above:
x = h, y = h - 5
```
So this will yield `(h, h-5)`which is also incorrect. 
Unless my math is failing me (which is very much a possibility :^) ), I do not see how achieving the rotation before the sync is possible. So this means that we do need the guard anyways since on the `INPUT_BTN_TOUCH` event with `value=0` we would pass again through the rotation code. At the point - yeah one could factor out the inversion I guess but this makes it more complicated imo, therefore I reverted back to the first version as per your suggestion :^)